census_view.cpp

```diff
@@ -87,700 +89,536 @@ std::string insert_spaces_between_words(std::string const& s)
+    char const* type() const override
     {
-        return new(wx) IntSpinRenderer();
+        return typeid(table_int_range_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        return new(wx) wxGridCellNumberRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeIntEditor();
     }
 };
 
-class renderer_double_range_converter : public renderer_range_converter
+// class table_double_range_converter
+
+class table_double_range_converter : public renderer_type_converter
 {
   public:
-    wxDataViewRenderer* create_renderer(any_member<Input> const&) const override
+    char const* type() const override
     {
-        return new(wx) DoubleRangeRenderer();
+        return typeid(table_double_range_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        // Use wxGridCellStringRenderer instead of wxGridCellFloatRenderer to
+        // keep the number format as is.
+        return new(wx) wxGridCellStringRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeDoubleEditor();
     }
 };
 
-class renderer_date_converter : public renderer_range_converter
+// class table_date_converter
+
+class table_date_converter : public renderer_type_converter
 {
   public:
-    wxDataViewRenderer* create_renderer(any_member<Input> const&) const override
+    wxString to_renderer_value(std::string const& value) const override
+    {
+        auto const date{ConvertDateToWx(value_cast<calendar_date>(value))};
+        return date.FormatDate();
+    }
+
+    std::string from_editor_value(wxString const& value) const override
+    {
+        wxDateTime date;
+        auto date_parse_ok = date.ParseISODate(value);
+        LMI_ASSERT(date_parse_ok);
+
+        return value_cast<std::string>(ConvertDateFromWx(date));
+    }
+
+    char const* type() const override
     {
-        return new(wx) DateRenderer();
+        return typeid(table_date_converter).name();
+    }
+
+    wxGridCellRenderer* create_renderer() const override
+    {
+        return new(wx) wxGridCellDateRenderer();
+    }
+
+    wxGridCellEditor* create_editor() const override
+    {
+        return new(wx) RangeDateEditor();
     }
 };
 
-// class renderer_fallback_converter
+// class table_fallback_converter
 
-class renderer_fallback_converter : public renderer_type_converter
+class table_fallback_converter : public renderer_type_converter
 {
   public:
-    wxVariant to_variant
-        (any_member<Input> const& x
-        ,Input             const&
-        ,std::string       const&
-        ) const override
+    wxString to_renderer_value(std::string const& value) const override
     {
         // Strings containing new line characters are currently not displayed
         // correctly by wxDataViewCtrl, so display the value on a single line
```

Comment:
`wxDVC` mention is out of date. I also wonder if this entire class is not unnecessary now, I think `wxGrid` should handle new lines just fine.